### PR TITLE
Chapter 14 : Following User

### PR DESCRIPTION
**Pull request by DaudKhadijah01**

### CODE REVIEW

**Comment by DaudKhadijah01** [comment]
ready

File: app/assets/stylesheets/relationships.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Relationships controller here.
+// They will automatically be included in application.css.
```
Comment (ngocvt-0484) [inline app/assets/stylesheets/relationships.scss]: delete this file because not use

File: app/models/relationship.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class Relationship < ApplicationRecord
+  belongs_to :follower, class_name: "User"
```
Comment (ngocvt-0484) [inline app/models/relationship.rb]:
```suggestion
  belongs_to :follower, class_name: User.name
```

File: app/helpers/relationships_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module RelationshipsHelper
+end
```
Comment (ngocvt-0484) [inline app/helpers/relationships_helper.rb]: delete

File: app/models/user.rb [hunk 4]
```diff
@@ -69,7 +93,15 @@ def password_reset_expired?
   # Defines a proto-feed.
   # See "Following users" for the full implementation.
   def feed
-    Micropost.where("user_id = ?", id)
+    following_ids = "SELECT followed_id FROM relationships
+                     WHERE follower_id = :user_id"
+    Micropost.where("user_id IN (#{following_ids})
+                     OR user_id = :user_id", user_id: id)Micropost.where("user_id IN (:following_ids) OR user_id = :user_id", following_ids: following_ids, user_id: id)
```
Comment (ngocvt-0484) [inline app/models/user.rb]: ???
check again
Micropost.where two times

**Comment by DaudKhadijah01** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**